### PR TITLE
IPAM BMSet scale functionality

### PR DESCRIPTION
**Pull request by abays**
Currently contains fixes for duplicate hostname/IP issues, as well as another problem involving a stuck reconciliation loop for BaremetalSets.  However, it does not clean-up the respective hostnames/IPs in OvercloudIpSet status nor OvercloudNet status during BaremetalSet scale-down.  We are leaving it this way for now.

### CODE REVIEW

File: controllers/overcloudipset_controller.go [hunk 1]
```diff
@@ -122,20 +122,35 @@ func (r *OvercloudIPSetReconciler) Reconcile(req ctrl.Request) (ctrl.Result, err
...
-			// record the reservation on the OvercloudNet
-			network.Status.Reservations = reservation
-			err = r.Client.Status().Update(context.TODO(), network)
-			if err != nil {
-				r.Log.Error(err, "Failed to update OvercloudNet status %v")
-				return ctrl.Result{}, err
+			// Do we already have a reservation for this hostname on the network?
+			for _, reservation := range network.Status.Reservations {
+				if reservation.Hostname == hostname {
+					reservationIP = reservation.IP
```
Comment (stuggi) [inline controllers/overcloudipset_controller.go:130]: I think with this we want to remove the check in L103-L105 and return the already existing reservation here.
```
			if instance.Status.HostIPs[hostname].IPAddresses[netName] != "" {
				continue
			}
```
I have seen issues with our current version and the above check/continue as ctrlplanCidr might not get set and added to the environment file.

Comment (abays) [reply controllers/overcloudipset_controller.go:130]: Yes, I think that might be right.  If the `hostname` is already present in the OvercloudIPSet status, then we can assume the same reservation is also already stored in the OvercloudNet.  But should we return, or do something else (other than `continue`)?  Do we want the `EnsureConfigMaps` logic to be processed?  A return would skip that logic.

Comment (stuggi) [reply controllers/overcloudipset_controller.go:130]: atm I think we want it processed with the existing information

Comment (dprince) [reply controllers/overcloudipset_controller.go:130]: Agree that you can probably drop the check at line 103 with this new approach. It probably doesn't hurt to recreate the ConfigMaps

Comment (abays) [reply controllers/overcloudipset_controller.go:130]: Sounds good.  I will remove it and run the change through my tests.

**Pull request merged**